frontend/badges/cppCourseGraduate/query.sql

```diff
@@ -0,0 +1,44 @@
+SELECT
+    p.user_id
+FROM (
+    SELECT
+        i.user_id,
+        MAX(r.contest_score) AS max_contest_score,
+        psp.problem_id
+    FROM
+        Assignments a
+    INNER JOIN
+        Problemsets ps ON a.problemset_id = ps.problemset_id
+    INNER JOIN
+        Problemset_Problems psp ON psp.problemset_id = ps.problemset_id
+    INNER JOIN
+        Submissions s ON s.problem_id = psp.problem_id AND
+        s.problemset_id = a.problemset_id
+    INNER JOIN
+        Runs r ON r.run_id = s.current_run_id
+    INNER JOIN
+        Identities i ON i.identity_id = s.identity_id
+    INNER JOIN
+        Courses c ON c.course_id = a.course_id
+    WHERE
+        c.alias = 'introduccion_a_cpp'
```

Comment:
```suggestion
        c.alias = 'introduccion_a_cpp' AND
        i.user_id IS NOT NULL
```

sabes qué? creo que esto es lo que falta para que Psalm esté contento.